server/server.go

```diff
@@ -347,7 +350,45 @@ func (ss *subStore) Remove(sub *subState, force bool) {
 		// for which we don't have substore lock held.
 		qs.Lock()
 		qs.subs, _ = sub.deleteFromList(qs.subs)
+		empty := len(qs.subs) == 0
 		qs.Unlock()
+		// If it was the last being removed, just remove the
+		// queue group from the subStore map of queue groups.
+		if empty {
+			delete(ss.qsubs, qgroup)
+		} else if hasPending {
+			// We need to "transfer" the possibly unacknowledged messages
+			// from the leaving member to the remaining one(s).
+			qs.Lock()
+			// For this case, we will do a simply round-robin, giving
+			// a message to each subscriber, etc..
+			numQSubs := len(qs.subs)
+			idx := 0
+			sub.RLock()
+			for _, m := range sub.acksPending {
+				// Get a remaning queue subscriber.
+				qsub := qs.subs[idx]
+				qsub.Lock()
+				// Give it one of the message.
+				qsub.acksPending[m.Sequence] = m
+				// Make sure we set its ack timer if none already set.
```

Comment:
Realized that's not enough. The message need to be added to the subscription store (persisted on disk for file stores). Need to revisit this.